### DISCUSSION TITLE
Ask if possible to confirm a crash with  an example code with walrus use,  on V1.19 ?

### DISCUSSION THREAD

**Discussion by fca1** [micropython/micropython#10879] [General]
Hi All,!

I've encourated a strange behavior with micropython 1.19 (crash and reset), my target is an ESP32 with spiram. 
The piece of code to attempt to reproduce that, is very tiny. This code has been tested with cpython, without any problems. 

Please, if someone could confirm if there is really a problem ? Thanks a lot in advance, 

def for_test_purpose():
    lst= [ 0,1]
    v=0
    x=1  # crash if  this line is present. 
    print([v := i for i in lst if i > v])

if __name__ == "__main__":
    for_test_purpose()

**Comment by scruss** [comment]
Crash confirmed on v1.19.1 with ESP32, ESP8266 and RP2040. 
v1.19.1 does not, however, crash on pyboard

Your code, reformatted to run:

``` python3
def for_test_purpose():
    lst = [0, 1]
    v = 0
    x = 1  # crash if this line is present.
    print([v := i for i in lst if i > v])

if __name__ == "__main__":
    for_test_purpose()
```
Might be better to raise this as an issue

**Reply by ricksorensen** [reply]
Same on SEEED_XIAO_SAMD21, firmware compiled this week.  Function hangs the system, no exceptions reported. Removing `x=1` and function runs okay.
`3.4.0; MicroPython v1.19.1-891-g8d9a7fd22 on 2023-02-22`

**Comment by robert-hh** [comment]
Ran that on a SAMD51 with a fresh build in the debugger. 
MicroPython v1.19.1-909-g36ae5d3e8 on 2023-03-01; ItsyBitsy M4 Express with SAMD51G19A

This is the error window showing the flags:

![grafik](https://user-images.githubusercontent.com/12476868/222073107-95be16a5-a32e-42d5-83f5-3a47a8c659e2.png)
And this is the call stack:

![grafik](https://user-images.githubusercontent.com/12476868/222073258-cac4ceea-bef9-4ac7-a05c-62182a5d6205.png)

**Comment by robert-hh** [comment]
I converted that into an issue.
